config/yaml.go

```diff
@@ -329,36 +325,34 @@ func unmarshalYAMLValue(reader io.ReadCloser, value interface{}) error {
 //
 // TODO: what if someone wanted a literal ${FOO} in config? need a small escape hatch
 func interpolateEnvVars(data []byte) ([]byte, error) {
-	var errs []string
-	data = _interpolationRegex.ReplaceAllFunc(data, func(in []byte) []byte {
-		// remove ${ and }, just look inside the braces
-		valAndMaybeDefault := in[2 : len(in)-1]
-
-		// ${VALUE:DEFAULT} -> VALUE, DEFAULT
-		// ${VALUE} -> VALUE, empty default
-		split := bytes.Split(valAndMaybeDefault, _envSeparator)
-		key := string(split[0])
-
-		if envVal, ok := os.LookupEnv(key); ok {
-			return []byte(envVal)
+	// Is this conversion ok?
+	str := string(data)
+	errs := []string{}
```

Comment:
I meant to use errors.Wrap(), that is going to have a message for values that don't have valid defaults. It would require an additional if though, so it is up to you.